runtime/lua/vim/lsp/protocol.lua

```diff
@@ -789,6 +789,7 @@ end
 --- | vim.lsp.protocol.Method.ServerToClient
 
 -- Generated by gen_lsp.lua, keep at end of file.
+--- @deprecated Use `vim.lsp.protocol.Method` instead.
```

Comment:
The changes in this PR are good. If we find that users have some need to iterate all of the method names, we might choose to un-deprecate this. That doesn't change the general guidance that source code should use the type literals for all other cases.